auction-server/src/opportunity/service/mod.rs

```diff
@@ -100,19 +116,35 @@ impl ConfigEvm {
 
 // NOTE: Do not implement debug here. it has a circular reference to auction_service
 pub struct ConfigSvm {
-    pub auction_service:         RwLock<Option<auction_service::Service<Svm>>>,
+    pub auction_service:         RwLock<Option<AuctionService<Svm>>>,
     pub rpc_client:              RpcClient,
     pub accepted_token_programs: Vec<Pubkey>,
     pub ordered_fee_tokens:      Vec<Pubkey>,
 }
 
 impl ConfigSvm {
     // TODO Move these to config trait?
-    pub async fn inject_auction_service(&self, service: auction_service::Service<Svm>) {
+    #[allow(unused_variables)]
```

Comment:
okay so this style of injection and mockall are definietly not playing well together

Reply:
you should be able to leverage #[double] here 
make it :
`pub async fn inject_auction_service(&self, service: crate::auction::service::Service<Svm>)
`?

Reply:
yeah but that brings a lot of other code to its knees, will try to figure out a neater solution to the circular dep here